docs/iam-usage.rst

```diff
@@ -0,0 +1,144 @@
+Using IAM For Users
+===================
+
+.. note:: This document does not explain how Google Cloud IAM works,
+   for that, please check out `the docs <https://cloud.google.com/iam/docs/>`_.
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Methods
+----------------------
+
+.. note:: All policy editing methods except ``set_policy`` have an optional
+   ``retries`` keyword argument that takes the number of times to retry in the
+   even of an interrupted transaction
+
+>>> resource.add_roles(iam.user('alice@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+True
+
+The first argument is one of the IAM "Member" types:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google-For-Work domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated user
+- ``iam.ALL_USERS``
+
+Note that all of these are convenience wrappers around strings. See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+All following arguments are roles to added to the specified member. Roles are simply strings that correspond to sets of permissions.
+The ``iam`` module provides a (possibly incomplete) list of curated roles.
+You can acquire the complete list by running ``gcloud iam list-grantable-roles //fully/qualified/resource/path``, and use the
+names provided there in place of ``iam.roles.XXXX.name`` wherever the latter is used in this doc.
+
+This method returns ``True`` if the change to the resource's policy was made successfully, and returns ``False`` otherwise
+(such as in the case of an update being made during this update).
+
+>>> resource.remove_roles(iam.group('devs@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+False
+
+Same as above, but removes the specified roles from the specified member.
+
+>>> resource.add_members(iam.roles.OWNER.name, iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com'))
+True
+
+Same as above, but the first argument is a role, and all subsequent arguments are members to which that role should be added.
+
+>>> resource.remove_members(iam.roles.OWNER.name, iam.ALL_USERS)
+True
+
+Same as above, but removes all listed members from the specified role.
+
+>>> def remove_fn(member):
```

Comment:
Ok, I was just thinking if that's a common use case, then it might be nice to make a helper function for end users.